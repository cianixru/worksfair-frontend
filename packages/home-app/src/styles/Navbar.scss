
.user-menu {
    position: relative;
    display: inline-block;
    cursor: pointer;
    margin-right: 25px;
    top: 9px;
    vertical-align: top;
    padding-left: 25px;
    margin-left: 25px
}

.user-menu .user-name {
    transition: .2s
}

.user-menu .user-name:after {
    content: "\f107";
    font-family: fontawesome;
    transition: .2s;
    transform: rotate(0);
    display: inline-block;
    margin-left: 6px;
    color: #aaa
}

.user-menu.active .user-name:after,
.user-menu:hover .user-name:after,
.user-menu.active .user-name,
.user-menu:hover .user-name {
    color: #66676b
}

.user-menu.active .user-name:after {
    transform: rotate(180deg)
}

.user-name span img {
    width: 100%;
    display: inline-block;
    border-radius: 50%;
    image-rendering: -webkit-optimize-contrast
}

.user-name span {
    display: inline-block;
    width: 32px;
    height: 32px;
    margin-right: 10px;
    border-radius: 50%;
    position: absolute;
    left: -22px;
}

.user-menu ul {
    text-align: left;
    position: absolute;
    top: 45px;
    right: 0;
    list-style: none;
    background-color: #fff;
    box-shadow: 0 0 10px 0 rgba(0, 0, 0, .12);
    border-radius: 4px;
    display: inline-block;
    width: 180px;
    font-size: 15px;
    transform: translate3d(0, 15px, 0);
    padding: 12px 10px;
    box-sizing: border-box;
    transition: .25s;
    visibility: hidden;
    opacity: 0;
    z-index: 110
}

.user-menu ul li a {
    padding: 6px 15px;
    line-height: 22px;
    display: inline-block;
    color: #696969;
    transition: .2s
}

.user-menu ul li a:hover {
    color: #444
}

.user-menu ul li a i {
    width: 20px;
    position: relative;
    display: inline-block
}

.user-menu.active ul {
    visibility: visible;
    opacity: 1;
    transform: translate3d(0, 0, 0)
}
