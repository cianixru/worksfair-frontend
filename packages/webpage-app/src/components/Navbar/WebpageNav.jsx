import React, { Component } from 'react';
import { withRouter } from 'react-router-dom';
import PropTypes from 'prop-types';
import dotenv from 'dotenv';

import SocialIcon from '../../atoms/SocialIcon';

dotenv.config();

class WebpageNav extends Component {
  componentDidMount() {
    const navBurger = this.navBurger.current;
    navBurger.addEventListener('click', this.toggleNavbar);
    window.addEventListener('scroll', this.scrollHandler);
  }

  componentWillUnmount() {
    const navBurger = this.navBurger.current;
    navBurger.removeEventListener('click', this.toggleNavbar);
    window.removeEventListener('scroll', this.scrollHandler);
  }

   navBurger = React.createRef()

   navBar = React.createRef()

   navMenu = React.createRef()

  scrollHandler = (event) => {
    const navbar = this.navBar.current;
    const { pageYOffset } = event.currentTarget;
    if (pageYOffset >= 70) {
      navbar.className = 'navbar is-fixed-top';
    } else {
      navbar.className = 'navbar';
    }
  }

  toggleNavbar = (event) => {
    event.preventDefault();

    const navMenu = this.navMenu.current;
    navMenu.classList.toggle('is-active');
  }

  render() {
    const { webpage, location } = this.props;
    const { REACT_APP_URL } = process.env;

    return (
      <nav
        className="navbar webnav"
        role="navigation"
        aria-label="main navigation"
        data-testid="WebpageNavbar"
        ref={this.navBar}
      >
        <div className="navbar-brand">
          <a
            className="navbar-item"
            href={`${location.pathname}#home`}>
            { webpage && webpage.logo
              ? <img src={webpage.logo} width="112" height="28" alt="logo" />
              : <h2 className="webpage-title">
                <strong>{ webpage && webpage.title }</strong>
              </h2>
            }
          </a>

          <a role="button"
            className="navbar-burger burger"
            aria-label="menu"
            aria-expanded="false"
            data-target="navbarBasicExample"
            ref={this.navBurger}
            href="/">

            <span aria-hidden="true" />
            <span aria-hidden="true" />
            <span aria-hidden="true" />
          </a>
        </div>

        <div
          className="navbar-menu"
          ref={this.navMenu}>
          <div className="navbar-start">
            <a
              className="navbar-item is-tab"
              href={`${location.pathname}#home`}>
              Home
            </a>
            <a
              className="navbar-item is-tab"
              href={`${location.pathname}#about`}>
              About
            </a>
            <a
              className="navbar-item is-tab"
              href={`${location.pathname}#contact`}>
              Contact
            </a>
            <a
              className="navbar-item is-tab"
              href={`${location.pathname}#offerings`}>
              Services/Products
            </a>
          </div>
          <div className="navbar-end">
            <div className="navbar-item">
              <a
                className="button is-outlined is-rounded"
                href={REACT_APP_URL}
              >
                Go to Worksfair
              </a>
            </div>
            { webpage && <div className="level is-mobile">
              <div className="level-left">
                {
                  webpage.facebook
                    && <a
                      href={webpage.facebook}
                      className="level-item"
                    >
                      <SocialIcon icon="fa fa-facebook" />
                    </a>
                }
                {
                  webpage.twitter
                    && <a
                      href={webpage.twitter}
                      className="level-item"
                    >
                      <SocialIcon icon="fa fa-twitter" />
                    </a>
                }
                {
                  webpage.instagram
                    && <a
                      href={webpage.instagram}
                      className="level-item"
                    >
                      <SocialIcon icon="fa fa-instagram" />
                    </a>
                }
              </div>
            </div>
            }
          </div>
        </div>
      </nav>
    );
  }
}

WebpageNav.propTypes = {
  user: PropTypes.object,
  webpage: PropTypes.object,
  location: PropTypes.object,
};

export default withRouter(WebpageNav);
