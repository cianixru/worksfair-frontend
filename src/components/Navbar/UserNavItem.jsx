import React from 'react';
import { NavLink, withRouter } from 'react-router-dom';
import { resolve } from 'path';
import PropTypes from 'prop-types';

import avatar from '../../assets/worksfair-avatar.png';

const UserNavItem = ({ user, links }) => {
  return (
    <div className="user-menu">
      <div className="user-name ">
        <span>
          <img
            src={user.image_url ? user.image_url : avatar}
            alt="avatar" />
        </span>
        {`${user.first_name} ${user.last_name}`}
      </div>
      <ul>
        {links.map(link => (
          <li
            key={link.text}
            className={link.class}
          >
            <NavLink to={resolve(link.to)} activeClassName="active">
              <i className={link.icon} aria-hidden="true" />
              {link.text}
            </NavLink>
          </li>
        ))}
      </ul>
    </div>
  );
};

UserNavItem.propTypes = {
  user: PropTypes.object,
  links: PropTypes.array,
};

export default withRouter(UserNavItem);
