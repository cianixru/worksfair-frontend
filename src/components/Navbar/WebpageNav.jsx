import React, { Component } from 'react';
import { Link, withRouter } from 'react-router-dom';
import PropTypes from 'prop-types';

class WebpageNav extends Component {
  componentDidMount() {
    const navBurger = this.navBurger.current;
    navBurger.addEventListener('click', this.toggleNavbar);
    window.addEventListener('scroll', this.scrollHandler);
  }

  componentWillUnmount() {
    const navBurger = this.navBurger.current;
    navBurger.removeEventListener('click', this.toggleNavbar);
    window.removeEventListener('scroll', this.scrollHandler);
  }

   navBurger = React.createRef()

   navBar = React.createRef()

   navMenu = React.createRef()

  scrollHandler = (event) => {
    const navbar = this.navBar.current;
    const { pageYOffset } = event.currentTarget;
    if (pageYOffset >= 70) {
      navbar.className = 'navbar is-fixed-top';
    } else {
      navbar.className = 'navbar';
    }
  }

  toggleNavbar = (event) => {
    event.preventDefault();

    const navMenu = this.navMenu.current;
    navMenu.classList.toggle('is-active');
  }

  render() {
    const { webpage, location } = this.props;

    return (
      <nav
        className="navbar webnav"
        role="navigation"
        aria-label="main navigation"
        data-testid="WebpageNavbar"
        ref={this.navBar}
      >
        <div className="navbar-brand">
          <a
            className="navbar-item"
            href={`${location.pathname}#home`}>
            { webpage && webpage.logo
              ? <img src={webpage.logo} width="112" height="28" alt="logo" />
              : <h2 className="webpage-title">
                <strong>{ webpage && webpage.title }</strong>
              </h2>
            }
          </a>

          <a role="button"
            className="navbar-burger burger"
            aria-label="menu"
            aria-expanded="false"
            data-target="navbarBasicExample"
            ref={this.navBurger}
            href="/">

            <span aria-hidden="true" />
            <span aria-hidden="true" />
            <span aria-hidden="true" />
          </a>
        </div>

        <div
          className="navbar-menu"
          ref={this.navMenu}>
          <div className="navbar-start">
            <a
              className="navbar-item is-tab"
              href={`${location.pathname}#home`}>
              Home
            </a>
            <a
              className="navbar-item is-tab"
              href={`${location.pathname}#about`}>
              About
            </a>
            <a
              className="navbar-item is-tab"
              href={`${location.pathname}#contact`}>
              Contact
            </a>
            <a
              className="navbar-item is-tab"
              href={`${location.pathname}#offerings`}>
              Services/Products
            </a>
          </div>
          <div className="navbar-end">
            <div className="navbar-item">
              <Link
                className="button is-outlined is-rounded"
                to="/"
              >
                Create Free Webpage
              </Link>
            </div>
          </div>
        </div>
      </nav>
    );
  }
}

WebpageNav.propTypes = {
  user: PropTypes.object,
  webpage: PropTypes.object,
  location: PropTypes.object,
};

export default withRouter(WebpageNav);
