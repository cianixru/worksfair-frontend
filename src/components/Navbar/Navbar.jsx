import React, { Component } from 'react';
import { Link, withRouter } from 'react-router-dom';
import PropTypes from 'prop-types';

import logo from '../../assets/logo.png';
import UserNavItem from './UserNavItem';

class Navbar extends Component {
  componentDidMount() {
    const navBurger = this.navBurger.current;
    navBurger.addEventListener('click', this.toggleNavbar);
    window.addEventListener('scroll', this.scrollHandler);
  }

  componentWillUnmount() {
    const navBurger = this.navBurger.current;
    navBurger.removeEventListener('click', this.toggleNavbar);
    window.removeEventListener('scroll', this.scrollHandler);
  }

  navBurger = React.createRef()

  navBar = React.createRef()

  navMenu = React.createRef()

  scrollHandler = (event) => {
    const navbar = this.navBar.current;
    const { pageYOffset } = event.currentTarget;
    if (pageYOffset >= 100) {
      navbar.className = 'navbar is-fixed-top';
    } else {
      navbar.className = 'navbar';
    }
  }

  toggleNavbar = (event) => {
    event.preventDefault();

    const navMenu = this.navMenu.current;
    const navBurger = this.navBurger.current;
    navMenu.classList.toggle('is-active');
    navBurger.classList.toggle('is-active');
  }

  render() {
    const { user } = this.props;
    const links = [
      {
        text: 'Dashboard',
        to: `/dashboard/${user && user.username}/webpages`,
        class: 'navbar-item',
      },
      {
        text: 'Logout',
        to: '/logout',
        class: 'navbar-item',
      },
    ];

    return (
      <nav className="navbar"
        role="navigation"
        aria-label="main navigation"
        ref={this.navBar}
        data-testid=""
      >
        <div className="navbar-brand">
          <Link className="navbar-item" to="/home">
            <img src={logo} width="112" height="28" alt="logo" />
          </Link>

          <a role="button"
            className="navbar-burger burger"
            aria-label="menu"
            aria-expanded="false"
            data-target="navbarBasicExample"
            ref={this.navBurger}
            href="/">

            <span aria-hidden="true" />
            <span aria-hidden="true" />
            <span aria-hidden="true" />
          </a>
        </div>

        {
          window.location.pathname === '/signup'
            || window.location.pathname === '/login'
            ? null
            : (<div
              className="navbar-menu"
              ref={this.navMenu}>
              <div className="navbar-start">
                <Link className="navbar-item" to="/">
                  Home
                </Link>

                <Link className="navbar-item" to="/">
                  About
                </Link>
              </div>

              <div className="navbar-end">
                <div className="navbar-item">

                  { user
                    ? <UserNavItem
                      user={user}
                      links={links} />
                    : <div className="buttons">
                      <Link className="button is-white" to="/login">
                        Log in
                      </Link>
                      <Link className="button is-white" to="/signup">
                        Sign up
                      </Link>
                      <Link
                        className="button is-link is-outlined is-rounded"
                        to="/login"
                      >
                        Add Your Business for Free
                      </Link>
                    </div>
                  }
                </div>
              </div>
            </div>)
        }
      </nav>
    );
  }
}

Navbar.propTypes = {
  user: PropTypes.object,
};

export default withRouter(Navbar);
