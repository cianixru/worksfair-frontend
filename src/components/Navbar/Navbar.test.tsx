import React from 'react';
import { render } from 'react-testing-library';
import Navbar from './Navbar';

test('Test that the Footer displays', () => {
  const component = render(
    <Navbar />
  );
  expect(component).toMatchSnapshot();
});