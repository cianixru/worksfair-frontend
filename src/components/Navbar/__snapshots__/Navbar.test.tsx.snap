// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test that the Navbar displays 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        aria-label="main navigation"
        class="navbar"
        data-testid=""
        role="navigation"
      >
        <div
          class="navbar-brand"
        >
          <a
            class="navbar-item"
            href="/home"
          >
            <img
              alt="logo"
              height="28"
              src="logo.png"
              width="112"
            />
          </a>
          <a
            aria-expanded="false"
            aria-label="menu"
            class="navbar-burger burger"
            data-target="navbarBasicExample"
            href="/"
            role="button"
          >
            <span
              aria-hidden="true"
            />
            <span
              aria-hidden="true"
            />
            <span
              aria-hidden="true"
            />
          </a>
        </div>
        <div
          class="navbar-menu"
        >
          <div
            class="navbar-start"
          >
            <a
              class="navbar-item"
              href="/"
            >
              Home
            </a>
            <a
              class="navbar-item"
              href="/"
            >
              About
            </a>
          </div>
          <div
            class="navbar-end"
          >
            <div
              class="navbar-item"
            >
              <div
                class="buttons"
              >
                <a
                  class="button is-white"
                  href="/login"
                >
                  Log in
                </a>
                <a
                  class="button is-white"
                  href="/signup"
                >
                  Sign up
                </a>
                <a
                  class="button is-danger is-outlined is-rounded"
                  href="/"
                >
                  Create Website
                </a>
              </div>
            </div>
          </div>
        </div>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      aria-label="main navigation"
      class="navbar"
      data-testid=""
      role="navigation"
    >
      <div
        class="navbar-brand"
      >
        <a
          class="navbar-item"
          href="/home"
        >
          <img
            alt="logo"
            height="28"
            src="logo.png"
            width="112"
          />
        </a>
        <a
          aria-expanded="false"
          aria-label="menu"
          class="navbar-burger burger"
          data-target="navbarBasicExample"
          href="/"
          role="button"
        >
          <span
            aria-hidden="true"
          />
          <span
            aria-hidden="true"
          />
          <span
            aria-hidden="true"
          />
        </a>
      </div>
      <div
        class="navbar-menu"
      >
        <div
          class="navbar-start"
        >
          <a
            class="navbar-item"
            href="/"
          >
            Home
          </a>
          <a
            class="navbar-item"
            href="/"
          >
            About
          </a>
        </div>
        <div
          class="navbar-end"
        >
          <div
            class="navbar-item"
          >
            <div
              class="buttons"
            >
              <a
                class="button is-white"
                href="/login"
              >
                Log in
              </a>
              <a
                class="button is-white"
                href="/signup"
              >
                Sign up
              </a>
              <a
                class="button is-danger is-outlined is-rounded"
                href="/"
              >
                Create Website
              </a>
            </div>
          </div>
        </div>
      </div>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Test that the Navbar displays with userNav 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        aria-label="main navigation"
        class="navbar"
        data-testid=""
        role="navigation"
      >
        <div
          class="navbar-brand"
        >
          <a
            class="navbar-item"
            href="/home"
          >
            <img
              alt="logo"
              height="28"
              src="logo.png"
              width="112"
            />
          </a>
          <a
            aria-expanded="false"
            aria-label="menu"
            class="navbar-burger burger"
            data-target="navbarBasicExample"
            href="/"
            role="button"
          >
            <span
              aria-hidden="true"
            />
            <span
              aria-hidden="true"
            />
            <span
              aria-hidden="true"
            />
          </a>
        </div>
        <div
          class="navbar-menu"
        >
          <div
            class="navbar-start"
          >
            <a
              class="navbar-item"
              href="/"
            >
              Home
            </a>
            <a
              class="navbar-item"
              href="/"
            >
              About
            </a>
          </div>
          <div
            class="navbar-end"
          >
            <div
              class="navbar-item"
            >
              <div
                class="buttons"
              >
                <a
                  class="button is-white"
                  href="/login"
                >
                  Log in
                </a>
                <a
                  class="button is-white"
                  href="/signup"
                >
                  Sign up
                </a>
                <a
                  class="button is-danger is-outlined is-rounded"
                  href="/"
                >
                  Create Website
                </a>
              </div>
            </div>
          </div>
        </div>
      </nav>
    </div>
    <div>
      <nav
        aria-label="main navigation"
        class="navbar"
        data-testid=""
        role="navigation"
      >
        <div
          class="navbar-brand"
        >
          <a
            class="navbar-item"
            href="/home"
          >
            <img
              alt="logo"
              height="28"
              src="logo.png"
              width="112"
            />
          </a>
          <a
            aria-expanded="false"
            aria-label="menu"
            class="navbar-burger burger"
            data-target="navbarBasicExample"
            href="/"
            role="button"
          >
            <span
              aria-hidden="true"
            />
            <span
              aria-hidden="true"
            />
            <span
              aria-hidden="true"
            />
          </a>
        </div>
        <div
          class="navbar-menu"
        >
          <div
            class="navbar-start"
          >
            <a
              class="navbar-item"
              href="/"
            >
              Home
            </a>
            <a
              class="navbar-item"
              href="/"
            >
              About
            </a>
          </div>
          <div
            class="navbar-end"
          >
            <div
              class="navbar-item"
            >
              <div
                class="user-menu"
              >
                <div
                  class="user-name "
                >
                  <span>
                    <img
                      alt="avatar"
                      src="worksfair-avatar.png"
                    />
                  </span>
                  eod okafor
                </div>
                <ul>
                  <li>
                    <a
                      href="/dashboard/Theookafor2"
                    >
                      <i
                        aria-hidden="true"
                        class="fa fa-tachometer"
                      />
                      Dashboard
                    </a>
                  </li>
                  <li>
                    <a
                      href="/logout"
                    >
                      <i
                        aria-hidden="true"
                        class="fa fa-power-off"
                      />
                      Logout
                    </a>
                  </li>
                </ul>
              </div>
            </div>
          </div>
        </div>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      aria-label="main navigation"
      class="navbar"
      data-testid=""
      role="navigation"
    >
      <div
        class="navbar-brand"
      >
        <a
          class="navbar-item"
          href="/home"
        >
          <img
            alt="logo"
            height="28"
            src="logo.png"
            width="112"
          />
        </a>
        <a
          aria-expanded="false"
          aria-label="menu"
          class="navbar-burger burger"
          data-target="navbarBasicExample"
          href="/"
          role="button"
        >
          <span
            aria-hidden="true"
          />
          <span
            aria-hidden="true"
          />
          <span
            aria-hidden="true"
          />
        </a>
      </div>
      <div
        class="navbar-menu"
      >
        <div
          class="navbar-start"
        >
          <a
            class="navbar-item"
            href="/"
          >
            Home
          </a>
          <a
            class="navbar-item"
            href="/"
          >
            About
          </a>
        </div>
        <div
          class="navbar-end"
        >
          <div
            class="navbar-item"
          >
            <div
              class="user-menu"
            >
              <div
                class="user-name "
              >
                <span>
                  <img
                    alt="avatar"
                    src="worksfair-avatar.png"
                  />
                </span>
                eod okafor
              </div>
              <ul>
                <li>
                  <a
                    href="/dashboard/Theookafor2"
                  >
                    <i
                      aria-hidden="true"
                      class="fa fa-tachometer"
                    />
                    Dashboard
                  </a>
                </li>
                <li>
                  <a
                    href="/logout"
                  >
                    <i
                      aria-hidden="true"
                      class="fa fa-power-off"
                    />
                    Logout
                  </a>
                </li>
              </ul>
            </div>
          </div>
        </div>
      </div>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
